lstchain/calib/camera/calibration_calculator.py

```diff
@@ -0,0 +1,258 @@
+"""
+Component for the estimation of the calibration coefficients  events
+"""
+
+
+from abc import abstractmethod
+import numpy as np
+import os
+from ctapipe.core import Component
+from ctapipe.core import traits
+from ctapipe.core.traits import Unicode, Float, List
+from lstchain.calib.camera.flatfield import FlatFieldCalculator
+from lstchain.calib.camera.pedestals import PedestalCalculator
+
+
+__all__ = [
+    'CalibrationCalculator',
+    'LSTCalibrationCalculator'
+]
+
+
+class CalibrationCalculator(Component):
+    """
+    Parent class for the camera calibration calculators.
+    Fills the MonitoringCameraContainer on the base of calibration events
+
+    Parameters
+    ----------
+
+    flatfield_calculator: lstchain.calib.camera.flatfield
+         The flatfield to use. If None, then FlatFieldCalculator
+            will be used by default.
+
+    pedestal_calculator: lstchain.calib.camera.pedestal
+         The pedestal to use. If None, then
+           PedestalCalculator will be used by default.
+
+    kwargs
+
+    """
+
+    pedestal_product = traits.enum_trait(
+        PedestalCalculator,
+        default='PedestalIntegrator'
+    )
+
+    flatfield_product = traits.enum_trait(
+        FlatFieldCalculator,
+        default='FlasherFlatFieldCalculator'
+    )
+
+    classes = List([
+                    FlatFieldCalculator,
+                    PedestalCalculator
+                    ]
+                   + traits.classes_with_traits(FlatFieldCalculator)
+                   + traits.classes_with_traits(PedestalCalculator)
+
+                   )
+
+    def __init__(
+        self,
+        **kwargs
+
+    ):
+
+        """
+        Parent class for the camera calibration calculators.
+        Fills the MonitoringCameraContainer on the base of calibration events
+
+        Parameters
+        ----------
+
+        flatfield_calculator: lstchain.calib.camera.flatfield
+             The flatfield to use. If None, then FlatFieldCalculator
+                will be used by default.
+
+        pedestal_calculator: lstchain.calib.camera.pedestal
+             The pedestal to use. If None, then
+               PedestalCalculator will be used by default.
+
+        kwargs
```

Comment:
What are kwargs used here for? And is it safe to assume that both the FlatField and the PedestalCalculator get the same kwargs?

Reply:
It is all the trailers passed the CalibrationCalculator. Each Component will look for each own trailets

Reply:
But raise errors for unknown ones.

You should explictly pass `config` and `parent` to this class and pass `parent=self` to all subcomponents.

Reply:
mmh, I had often problem with this stuff. For the moment, I used the same way of proceeding with all the calibration Components and it works. Let keep the suggested change as a coming improvement for  the lstchain update to the ctapipe master, ok?

Reply:
No, please let's do it right here. It is actually pretty simple. If you have a component consisting of subcomponents, pass `parent=self` to the subcomponents.

Reply:
No, please let's do it right here. It is actually pretty simple. If you have a component consisting of subcomponents, pass `parent=self` to the subcomponents.

Reply:
Hi @MaxNoe, 
I still have problems to get the all the trailets in the Components without kargs. I don't think
this change is fundamental for the coming release (which is still based on a "bastard" code). 
Actually the change should be performed also in the ctapipe calibration Components. Let do everything properly everywhere in the successive release (both in lstchain and in ctapipe) where also trailets will be correctly passed through a tool script